sdmetrics/single_table/detection/base.py

```diff
@@ -43,6 +43,37 @@ def _fit_predict(X_train, y_train, X_test):
         """Fit a classifier and then use it to predict."""
         raise NotImplementedError()
 
+    @staticmethod
+    def _drop_non_compute_columns(real_data, synthetic_data, metadata):
+        """Drop all columns that cannot be statistically modeled."""
+        transformed_real_data = real_data
+        transformed_synthetic_data = synthetic_data
+
+        if metadata is not None:
+            drop_columns = []
+            drop_columns.extend(get_alternate_keys(metadata))
+            if 'columns' in metadata:
+                for column in metadata['columns']:
+                    if ('primary_key' in metadata and
+                            (column == metadata['primary_key'] or
+                             column in metadata['primary_key'])):
+                        drop_columns.append(column)
+
+                    for field in metadata['columns'][column]:
+                        if field == 'sdtype':
+                            sdtype = metadata['columns'][column][field]
+                            if sdtype not in ['numerical', 'datetime', 'categorical']:
+                                drop_columns.append(column)
+
+                        if field == 'pii':
+                            if metadata['columns'][column][field]:
+                                drop_columns.append(column)
```

Comment:
for both of these I think you could just get the field value from the dict and check directly. Something like
```python
column_info = metadata['columns'].get(column, {})
sdtype = column_info.get('sdtype')
pii = column_info.get('pii')
if sdtype not in ['numerical', 'datetime', 'categorical'] or pii:
    drop_columns.append(column)

Reply:
Yes that is much cleaner, thanks!